apps/desktop/src/lib/components/contextmenu/ContextMenu.svelte

```diff
@@ -6,203 +6,165 @@
-		if (exceedsLeft) {
-			newMenuPosition.x = 0;
-		}
-
-		// Check if the menu exceeds the window's bottom edge
-		const exceedsBottom = newMenuPosition.y + contextMenuHeight > window.innerHeight;
-		if (exceedsBottom) {
+		if (newMenuPosition.x < 0) newMenuPosition.x = 0;
+		if (newMenuPosition.y + contextMenuHeight > window.innerHeight) {
 			newMenuPosition.y = window.innerHeight - contextMenuHeight - menuWindowEdgesOffset;
 		}
+		if (newMenuPosition.y < 0) newMenuPosition.y = 0;
 
-		// Check if the menu exceeds the window's top edge
-		const exceedsTop = newMenuPosition.y < 0;
-		if (exceedsTop) {
-			newMenuPosition.y = 0;
-		}
-
-		// Apply the new position
 		menuPosition = newMenuPosition;
 	}
 
-	function setAlignByTarget() {
-		if (!target) return;
-
+	function setAlignByTarget(target: HTMLElement) {
 		const targetBoundingRect = target.getBoundingClientRect();
 		let newMenuPosition = {
 			x: setHorizontalAlign(targetBoundingRect),
 			y: setVerticalAlign(targetBoundingRect)
 		};
 
-		// Check if the menu goes beyond the window's right edge
-		const exceedsRight = newMenuPosition.x + contextMenuWidth > window.innerWidth;
-		if (exceedsRight) {
+		// Adjust alignment to stay within the window
+		if (newMenuPosition.x + contextMenuWidth > window.innerWidth) {
 			horizontalAlign = horizontalAlign === 'right' ? 'left' : 'right';
 			newMenuPosition.x = setHorizontalAlign(targetBoundingRect);
 		}
-
-		// Check if the menu goes beyond the window's left edge
-		const exceedsLeft = newMenuPosition.x < 0;
-		if (exceedsLeft) {
+		if (newMenuPosition.x < 0) {
 			horizontalAlign = 'right';
 			newMenuPosition.x = setHorizontalAlign(targetBoundingRect);
 		}
-
-		// Check if the menu goes beyond the window's bottom edge
-		const exceedsBottom = newMenuPosition.y + contextMenuHeight > window.innerHeight;
-		if (exceedsBottom) {
+		if (newMenuPosition.y + contextMenuHeight > window.innerHeight) {
 			verticalAlign = verticalAlign === 'bottom' ? 'top' : 'bottom';
 			newMenuPosition.y = setVerticalAlign(targetBoundingRect);
 		}
-
-		// Check if the menu goes beyond the window's top edge
-		const exceedsTop = newMenuPosition.y < 0;
-		if (exceedsTop) {
+		if (newMenuPosition.y < 0) {
 			verticalAlign = 'bottom';
 			newMenuPosition.y = setVerticalAlign(targetBoundingRect);
 		}
 
-		// Apply the new position
 		menuPosition = newMenuPosition;
 	}
 
 	export function open(e?: MouseEvent, newItem?: any) {
-		if (!target) return;
-
-		if (newItem) {
-			item = newItem;
-		}
+		if (!(leftClickTrigger || rightClickTrigger)) return;
 
+		item = newItem ?? item;
 		isVisible = true;
-		onopen?.();
+		savedMouseEvent = e;
 
-		if (!openByMouse) {
-			setAlignByTarget();
-		}
-
-		if (openByMouse && e) {
-			savedMouseEvent = e;
-		}
+		onopen?.();
+		if (ontoggle) executeByTrigger(ontoggle);
 	}
 
 	export function close() {
 		if (!isVisible) return;
-
 		isVisible = false;
+		// debugger;
+		// leftClickTrigger?.style.removeProperty('background');
```

Comment:
I think we can delete these two lines :)

Reply:
Done :-)